doc/src/manual/memory-management.md

```diff
@@ -0,0 +1,203 @@
+# [Memory Management and Garbage Collection](@id man-memory-management)
+
+Julia uses automatic memory management through its built-in garbage collector (GC). This section provides an overview of how Julia manages memory and how you can configure and optimize memory usage for your applications.
+
+## [Garbage Collection Overview](@id man-gc-overview)
+
+Julia features a garbage collector with the following characteristics:
+
+* **Non-moving**: Objects are not relocated in memory during garbage collection
+* **Generational**: Younger objects are collected more frequently than older ones
+* **Parallel and partially concurrent**: The GC can use multiple threads and run concurrently with your program
+* **Mostly precise**: The GC accurately identifies object references with minimal conservative scanning
+
+The garbage collector automatically reclaims memory used by objects that are no longer reachable from your program, freeing you from manual memory management in most cases.
+
+## [Memory Architecture](@id man-memory-architecture)
+
+Julia uses a two-tier allocation strategy:
+
+* **Small objects** (currently ≤ 2032 bytes but may change): Allocated using a fast per-thread pool allocator
+* **Large objects** : Allocated directly through the system's `malloc`
+
+This hybrid approach optimizes for both allocation speed and memory efficiency, with the pool allocator providing fast allocation for the many small objects typical in Julia programs.
+
+## [System Memory Requirements](@id man-system-memory)
+
+### Swap Space
+
+Julia's garbage collector is designed with the expectation that your system has adequate swap space configured. The GC uses heuristics that assume it can allocate memory beyond physical RAM when needed, relying on the operating system's virtual memory management.
```

Comment:
I'm not sure this adds much, this is true for every program out there.

Reply:
it is, but HPC people don't agree so IMO it's worth pointing out.